discretize/tests.py

```diff
@@ -633,6 +637,110 @@ def Quadratic(x, return_g=True, return_H=True):
     return Quadratic
 
 
+# TESTS
+def dottest(
+    forward,
+    adjoint,
+    fwd_shape,
+    adj_shape,
+    fwd_complex=False,
+    adj_complex=False,
+    fwd_order="C",
+    adj_order="C",
+    clinear=True,
+    rtol=1e-6,
+    atol=0.0,
+    raise_error=False,
+    verb=False,
+):
+    r"""Dot product test for the forward operator and its adjoint operator.
+
+    Dot product test to verify the correctness of the adjoint operator
+    :math:`F^H` of the forward operator :math:`F`.
+
+    .. math::
+
+        \mathbf{v}^H ( \mathbf{F} \mathbf{u} ) =
+        ( \mathbf{F}^H \mathbf{v} )^H \mathbf{u}
+
+
+    Parameters
+    ----------
+    forward, adjoint : functions
+        Forward operator and its adjoint operator.
+
+    fwd_shape, adj_shape : int, tuple of int
+        Shapes of the vectors {u; v} passed to ``forward`` and ``adjoint``.
+
+    fwd_complex, adj_complex : bool, defaults: False, False
+        If True, complex vectors are passed to ``forward`` and ``adjoint``.
+
+    fwd_order, adj_order : string, defaults: 'C', 'C'
+        Used the ravel the output of the ``forward``/``adjoint`` if their shape
+        is higher than 1D.
+
+    clinear : bool, default: True
+        If operator is complex-linear (True) or real-linear (False).
+
+    rtol, atol : float, defaults: 1e-6, 0.0
+        Relative and absolute tolerance, used with :func:`numpy.isclose`.
+
+    raise_error : bool, default: False
+        If False, the result of the test is returned as boolean.
+        If True and the test fails, an AssertionError is raised.
+
+    verb : bool, default: False
+        Verbosity; if True, print result if passed.
+
+
+    Returns
+    -------
+    passed : bool, returned if raise_error=False
+        Result of the dot product test.
+
+    """
+
+    def random(size, iscomplex):
+        """Create random data of size and dtype of <size>."""
+        out = rng.standard_normal(int(np.real(size)))
+        if iscomplex:
+            out = out + 1j * rng.standard_normal(out.size)
+        return out
+
+    # Create random vectors u and v.
+    u = random(np.product(fwd_shape), fwd_complex)
+    v = random(np.product(adj_shape), adj_complex)
+
+    # Carry out dot product test.
+    fwd_u = forward(u.reshape(fwd_shape, order=fwd_order)).ravel(fwd_order)
+    adj_v = adjoint(v.reshape(adj_shape, order=adj_order)).ravel(adj_order)
+    if clinear:
+        lhs = np.vdot(v, fwd_u)  # lhs := v^H * (fwd * u)
+        rhs = np.vdot(adj_v, u)  # rhs := (adj * v)^H * u
+    else:
+        lhs = np.dot(v.real, fwd_u.real) + np.dot(v.imag, fwd_u.imag)
+        rhs = np.dot(adj_v.real, u.real) + np.dot(adj_v.imag, u.imag)
+
+    # Check if they are the same.
+    passed = np.isclose(rhs, lhs, rtol, atol)
+
+    # Verbosity and error.
+    if (not passed and raise_error) or verb:
```

Comment:
And then here, just remove the rest of the function.  Which, I believe would remove the use of the `verb` parameter.

Reply:
Goes to the main discussion - should it be a check or an assertion.

Reply:
I removed the `verb` parameter, and renamed `raise_error` to `assert_error` (True by default). As such, I kept the "verbose" part so it could either be used as an assertion in tests, or as whatever the users want it to returning the boolean test result.